Quick-Start/Anatomy_of_extension.md

```diff
@@ -1,418 +1,418 @@
+ 
+2.  **Define the metadata**: Open the Control file in a text editor and start by defining the metadata for your extension. This includes the following information: 
+ 
+    -   **comment**: A brief description of the extension. 
+    -   **default_version**: The default version number of the extension. 
+    -   **module_pathname**: The path to the extension's shared library file. 
+    -   **relocatable**: Specify whether the extension can be relocated to a different directory. 
+    -   **schema**: The name of the schema where the extension objects will be created. 
+    -   **requires**: List any dependencies your extension has on other extensions or specific PostgreSQL versions. 
+3.  **Add extension options**: If your extension has custom configuration options, you can define them using the `--->` syntax. Each option should include a name, a default value, a description, and any other necessary attributes. 
+ 
+4.  **Specify extension scripts**: In the Control file, you need to specify which scripts are associated with your extension. These scripts include the SQL script file(s) responsible for creating or modifying the extension's objects. Use the `script` keyword followed by the script type (`pre-install`, `install`, `post-install`, `pre-upgrade`, `upgrade`, `post-upgrade`, `pre-uninstall`, `uninstall`, `post-uninstall`) and the path to the script file. 
+ 
+5.  **Include documentation**: Provide relevant documentation or references related to your extension. This can include links to external resources, examples, or any additional information that helps users understand and use your extension effectively. 
+
+6.  **Save the Control file**: Once you have defined all the necessary metadata and instructions, save the Control file with the appropriate name and extension. 
+
+Writing a Control file helps developers by: 
+
+-   Providing a standardized format for organizing and documenting extension metadata. 
+-   Enabling users and administrators to easily understand the purpose and requirements of the extension. 
+-   Facilitating the installation, upgrade, and management of the extension by specifying dependencies and associated scripts. 
+-   Allowing customization through extension options. 
+-   Providing a reference point for documentation and additional resources. 
+
+Remember to follow the specific format and guidelines defined by PostgreSQL when writing the Control file. The Control file plays a crucial role in the successful deployment and usage of your PostgreSQL extension. 
+
+
+# 3) Code File: 
+Code File: A code file in the context of PostgreSQL extensions refers to the source code file that contains the implementation of the extension's functionality. It is typically written in the C programming language and is responsible for defining the extension's functions, data types, operators, and other database entities. 
```

Comment:
This is only true for extensions that depend on an external compiler. SQL, PL/pgSQL, PL/Perl, etc. extensions do not require a source file separate from the SQL file.